### PR TITLE
Storage profile detail

### PR DESCRIPTION
**Pull request by cangfengzhs**

### CODE REVIEW

File: src/storage/exec/StoragePlan.h [hunk 1]
```diff
@@ -79,6 +79,13 @@ class StoragePlan {
         CHECK_LT(idx, nodes_.size());
         return nodes_[idx].get();
     }
+    std::vector<RelNode<T>*> getNodes() {
```
Comment (critical27) [inline src/storage/exec/StoragePlan.h:82]: Could just return nodes_'s const reference is ok.

`getNode` method is only used in test case.

File: src/storage/index/LookupBaseProcessor.h [hunk 2]
```diff
@@ -84,6 +86,8 @@ class LookupBaseProcessor : public BaseProcessor<RESP> {
     // Save schemas when column is out of index, need to read from data
     std::vector<std::shared_ptr<const meta::NebulaSchemaProvider>> schemas_;
     std::vector<size_t>                                            deDupColPos_;
+    std::map<std::string, int64_t>                                  profile_detail_;
+    std::mutex                                                      profile_mut_;
```
Comment (critical27) [inline src/storage/index/LookupBaseProcessor.h:90]: Alignment.

File: src/storage/index/LookupBaseProcessor.inl [hunk 3]
```diff
@@ -490,6 +490,16 @@ LookupBaseProcessor<REQ, RESP>::buildPlanWithDataAndFilter(nebula::DataSet* resu
         return output;
     }
 }
-
+template <typename REQ, typename RESP>
+void LookupBaseProcessor<REQ, RESP>::profile_plan(StoragePlan<IndexID>& plan) {
+    auto nodes = plan.getNodes();
+    std::unique_lock<std::mutex> lck(profile_mut_);
```
Comment (critical27) [inline src/storage/index/LookupBaseProcessor.inl:496]: lock_guard is enough

File: src/storage/index/LookupProcessor.cpp [hunk 4]
```diff
@@ -128,6 +134,12 @@ void LookupProcessor::onProcessFinished() {
                        [this](const auto& col) { return context_->tagName_ + "." + col; });
     }
     resp_.set_data(std::move(resultDataSet_));
+    if (FLAGS_profile_storage_detail) {
```
Comment (critical27) [inline src/storage/index/LookupProcessor.cpp:137]: Consider move this part to `BaseProcessor::onFinished`. `profile_detail_` could be a member of `BaseProcessor`.

File: src/storage/index/LookupProcessor.cpp [hunk 5]
```diff
@@ -128,6 +134,12 @@ void LookupProcessor::onProcessFinished() {
...
+        this->result_.set_latency_detail_us({});
+        for (auto iter : profile_detail_) {
```
Comment (critical27) [inline src/storage/index/LookupProcessor.cpp:139]: Could use move to construct the may in rpc response.

**Review by critical27 — COMMENTED** [review]
Good job, generally LGTM, some code need to moved to BaseProcessor.

File: src/storage/query/GetNeighborsProcessor.h [hunk 6]
```diff
@@ -10,6 +10,7 @@
 #include "common/base/Base.h"
 #include <gtest/gtest_prod.h>
 #include "storage/query/QueryBaseProcessor.h"
+#include <map>
```
Comment (liuyu85cn) [inline src/storage/query/GetNeighborsProcessor.h:13]: already in Base.h

File: src/storage/BaseProcessor.h [hunk 7]
```diff
@@ -46,6 +46,12 @@ class BaseProcessor {
         }
 
         this->result_.set_latency_in_us(this->duration_.elapsedInUSec());
+        if (!profile_detail_.empty()) {
+            this->result_.set_latency_detail_us({});
```
Comment (critical27) [inline src/storage/BaseProcessor.h:50]: result_.set_latency_detail_us(std::move(profile_detail_)) will work I think

**Review by critical27 — APPROVED** [review]

**Comment by cangfengzhs** [comment]
Has been merged into nebula repo.

**Pull request closed without merging**